spec/system/candidate_interface/candidate_with_unsuccessful_application_applies_again_spec.rb

```diff
@@ -56,4 +68,26 @@ def and_i_click_on_start_now
   def then_i_see_a_copy_of_my_application
     expect(page).to have_content('Your new application is ready for editing')
   end
+
+  def when_i_select_a_course
+    given_courses_exist
+    and_the_suitability_to_work_with_children_feature_flag_is_on
+
+    click_link 'Course choices'
+    candidate_fills_in_course_choices
+  end
+
+  def and_submit_the_application
+    click_link 'Check and submit your application'
+    click_link 'Continue'
+    choose 'No' # "Is there anything else you would like to tell us?"
+
+    click_button 'Submit application'
+
+    @application = ApplicationForm.last
+  end
+
+  def then_application_is_submitted
+    expect(page).to have_content('Application successfully submitted')
```

Comment:
The key thing to test here is whether or not the application is now in the "application_complete" state, since the references are already completed.